acceptance/tests/connect/connect_helper.go

```diff
@@ -17,40 +17,85 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-const staticClientName = "static-client"
-const staticServerName = "static-server"
+const (
+	staticClientName = "static-client"
+	staticServerName = "static-server"
+)
 
-// ConnectInjectConnectivityCheck is a helper function used by the connect tests and cli smoke tests to test service
-// mesh connectivity.
-func ConnectInjectConnectivityCheck(t *testing.T, ctx environment.TestContext, cfg *config.TestConfig, secure bool, autoEncrypt bool, cli bool) {
-	helmValues := map[string]string{
-		"connectInject.enabled": "true",
+// ConnectHelper configures a Consul cluster for connect injection tests.
+// It also provides helper methods to exercise the connect functionality.
+type ConnectHelper struct {
+	// ClusterKind is the kind of Consul cluster to use (e.g. "Helm", "CLI").
+	ClusterKind consul.ClusterKind
 
-		"global.tls.enabled":           strconv.FormatBool(secure),
-		"global.tls.enableAutoEncrypt": strconv.FormatBool(autoEncrypt),
-		"global.acls.manageSystemACLs": strconv.FormatBool(secure),
-	}
+	// Secure configures the Helm chart for the test to use ACL tokens.
+	Secure bool
 
-	releaseName := helpers.RandomName()
-	var consulCluster consul.Cluster
-	if cli {
-		consulCluster = consul.NewCLICluster(t, helmValues, ctx, cfg, consul.CLIReleaseName)
-	} else {
-		consulCluster = consul.NewHelmCluster(t, helmValues, ctx, cfg, releaseName)
+	// AutoEncrypt configures the Helm chart for the test to use AutoEncrypt.
+	AutoEncrypt bool
+
+	// HelmValues are the additional helm values to use when installing or
+	// upgrading the cluster beyond connectInject.enabled, global.tls.enabled,
+	// global.tls.enableAutoEncrypt, global.acls.mangageSystemACLs which are
+	// set by the Secure and AutoEncrypt fields.
+	HelmValues map[string]string
+
+	// RelaseName is the name of the Consul cluster.
+	ReleaseName string
+
+	T   *testing.T
```

Comment:
Can `testing.T` be something that each of the methods receives as an argument? This way, one could send in a `retry.R` which adheres to the same interface in-case one of the methods needs to be run until it succeeds. Also, it allows for some context-based de-coupling of `T`.
eg. Setup(t *testing.T)

Reply:
Yeah, I like that better. I think it's more consistent with existing code too.